### PR TITLE
Use arm deploy

### PR DESCRIPTION
**Pull request by devoncarew**
- bump the engine revision
- download the new `android-arm-deploy` bucket
- wire up `flutter build apk` to use the right `android-arm` or `android-arm-deploy` artifacts bucket from the cache
- remove a print statement from the bin/flutter script - we take the same code path through either way

### CODE REVIEW

**Comment by devoncarew** [comment]
Sample output:

```
flutter --engine-src-path=/Users/devoncarew/projects/flutter/engine/src build apk
Building APK in develop mode (android_Debug)...
Warning: signing the APK using the debug keystore.
Built build/app.apk (8.61MB).
[~/flutter/flutter_sunflower] flutter build apk
Building APK in develop mode (android-arm)...
Warning: signing the APK using the debug keystore.
Built build/app.apk (8.25MB).
[~/flutter/flutter_sunflower] flutter build apk --deploy
Building APK in deploy mode (android-arm-deploy)...
Warning: signing the APK using the debug keystore.
Built build/app.apk (8.00MB).
```

**Comment by eseidelGoogle** [comment]
I absolutely love that you printed the binary size at the end of the build command.  "You make what you measure" and all that. :)

File: packages/flutter_tools/lib/src/commands/build_apk.dart [hunk 1]
```diff
@@ -227,51 +224,43 @@ class BuildApkCommand extends FlutterCommand {
...
     String abiDir = platform == TargetPlatform.android_arm ? 'armeabi-v7a' : 'x86_64';
-    artifactPaths = [
-      '$enginePath/third_party/icu/android/icudtl.dat',
-      '${config.buildDir}/gen/sky/shell/shell/classes.dex.jar',
-      '${config.buildDir}/gen/sky/shell/shell/shell/libs/$abiDir/libsky_shell.so',
-      '$enginePath/build/android/ant/chromium-debug.keystore',
+    String enginePath = tools.engineSrcPath;
+    String buildDir = tools.getEngineArtifactsDirectory(platform, buildVariant).path;
+
+    components.icuData = new File('$enginePath/third_party/icu/android/icudtl.dat');
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: I see.  This is part of @chinmaygarde's work to make the artifacts.zip buckets look like the out/Foo/dist and vice versa.  Silly to have to encode this here.

File: packages/flutter_tools/lib/src/commands/build_apk.dart [hunk 2]
```diff
@@ -227,51 +224,43 @@ class BuildApkCommand extends FlutterCommand {
...
+    components.jars = <File>[
+      new File(path.join(artifacts.path, 'classes.dex.jar'))
     ];
-    artifactPaths = artifactTypes.map((ArtifactType type) {
-      return ArtifactStore.getPath(ArtifactStore.getArtifact(
-        type: type,
-        targetPlatform: config.targetPlatform
-      ));
-    }).toList();
+    components.libSkyShell = new File(path.join(artifacts.path, 'libsky_shell.so'));
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: I don't understand why some paths use the deep .so vs. this one.  I suspect one is stripped and one is not, and that one of these paths may be wrong.  @chinmaygarde or @jason-simmons might know.

File: packages/flutter_tools/lib/src/commands/build_apk.dart [hunk 3]
```diff
@@ -333,7 +322,8 @@ int _buildApk(
     File apkShaFile = new File('$outputFile.sha1');
     apkShaFile.writeAsStringSync(calculateSha(finalApk));
 
-    printStatus('Generated APK to ${finalApk.path}.');
+    double size = finalApk.lengthSync() / (1024 * 1024);
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: Kinda lame. I love Dart's Duration class, I'm slightly surprised they don't have something similar for file sizes. :D

**Comment by eseidelGoogle** [comment]
This is a fantastic change.  lgtm.

File: packages/flutter_tools/lib/src/commands/build_apk.dart (hunk 1 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: Yes, hopefully this gets simplified over time.

File: packages/flutter_tools/lib/src/commands/build_apk.dart (hunk 3 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: `finalApk.lengthSync().asMB()` would be cool :)

**Pull request merged**